src/nnet3/nnet-test-utils.cc

```diff
@@ -793,7 +793,7 @@ void ComputeExampleComputationRequestSimple(
 
 static void GenerateRandomComponentConfig(std::string *component_type,
                                           std::string *config) {
-  int32 n = RandInt(0, 20);
+  int32 n = RandInt(0, 21);
```

Comment:
BTW, it can be helpful to do something to the code which should cause the tests to fail (e.g. scale the output by some constant factor in the propagate function, which should cause the derivative test to fail), and verify that it does fail.  This can catch situations where for some reason the testing code was never being invoked.